### PR TITLE
some nice updates

### PR DESCRIPTION
**Pull request by tabsl**
use guzzle
improve data response
add d-get-signature
documentation

### CODE REVIEW

File: src/Credentials.php [hunk 1]
```diff
@@ -2,10 +2,22 @@
 
 namespace DPN\DHLShipmentTracking;
 
+/**
```
Comment (cedricziel) [inline src/Credentials.php]: Can you please remove the empty class headers? Nobody needs those...

File: src/Credentials.php [hunk 2]
```diff
@@ -32,6 +44,8 @@ class Credentials
     public $tnt_password;
 
     /**
+     * The constructor ...
```
Comment (cedricziel) [inline src/Credentials.php]: Remove comment header pls

File: src/RequestBuilder.php [hunk 3]
```diff
@@ -2,16 +2,52 @@
 
 namespace DPN\DHLShipmentTracking;
 
+/**
+ * Class RequestBuilder
```
Comment (cedricziel) [inline src/RequestBuilder.php]: Same here

File: src/ShipmentTracking.php [hunk 4]
```diff
@@ -24,110 +48,125 @@ public function __construct(Credentials $credentials)
...
+    {
+        foreach ($data as $event) {
+            $events[] = $event['@attributes'];
+        }
+
+        return array_reverse($events);
+    }
+
+    /**
+     * @param        $operation
```
Comment (cedricziel) [inline src/ShipmentTracking.php]: When adding doc comments, please also add the types

File: src/ShipmentTracking.php [hunk 5]
```diff
@@ -24,110 +48,125 @@ public function __construct(Credentials $credentials)
...
 
     /**
-     * @param string $operation
-     * @param string $pieceCode
+     * @param        $operation
+     * @param        $pieceCode
      * @param string $language
-     * @return bool|string
+     *
+     * @return mixed
```
Comment (cedricziel) [inline src/ShipmentTracking.php]: Why is the type mixed? What is returned?

File: src/ShipmentTracking.php [hunk 6]
```diff
@@ -24,110 +48,125 @@ public function __construct(Credentials $credentials)
...
+     */
+    private function getArray($xml)
     {
-        $xml_obj = new \SimpleXMLElement($xml);
-        $level = 4;
-        $indent = 0; // current indentation level
-        $pretty = array();
+        $xml = simplexml_load_string($xml);
```
Comment (cedricziel) [inline src/ShipmentTracking.php:162]: Can we please also require `"ext-simplexml": "*"` in composer.json?

File: src/ShipmentTracking.php [hunk 7]
```diff
@@ -24,110 +48,125 @@ public function __construct(Credentials $credentials)
...
 
-        $response = file_get_contents($this->credentials->cig_endpoint . '?xml=' . urlencode($request), false, $context);
-
-        return $this->xmlpp($response);
+        return $res->getBody();
     }
 
-    public function xmlpp($xml, $html_output = false)
+    /**
+     * @param $xml
```
Comment (cedricziel) [inline src/ShipmentTracking.php]: Please adjust the doccomment so it matches the reality or discard it

**Review by cedricziel — CHANGES_REQUESTED** [review]
Thx for your contribution - that's really some nice changes you got there.

Please adjust the doccomments so we know the types that are exchanged and remove empty class headers as mentioned.

Thx!

**Comment by tabsl** [comment]
done ...

**Comment by tabsl** [comment]
Any updates?

**Comment by proudcommerce** [comment]
Any updates?

**Review by cedricziel — APPROVED** [review]

**Comment by cedricziel** [comment]
Thank you, @tabsl !

**Pull request merged**